src/modules/RA/Viewpoint/views/List.js

```diff
@@ -1,13 +1,77 @@
 import React from 'react';
-import { List } from 'react-admin';
+import {
+  Filter,
+  List,
+  SelectArrayInput,
+  SelectInput,
+  TextInput,
+} from 'react-admin';
 
 import CommonBulkActionButtons from '../../../../components/react-admin/CommonBulkActionButtons';
 import GridList from '../../../../components/react-admin/GridList';
+import { fetchFilterOptions } from '../../ra-modules';
+import { connectAppProvider } from '../../../../components/AppProvider';
 
+
+const ListFilters = props => {
+  const [selectInputs, setSelectInputs] = React.useState([]);
+
+  const setFilters = async () => {
+    const filterOptions = await fetchFilterOptions();
+    const terraOppSearchableProperties = props.properties;
+
+    const selectInputList = Object.keys(terraOppSearchableProperties).reduce(
```

Comment:
Can you add some comment to summary the next code block ?

Reply:
Yes! This is in fact a necessary comment.